tests/integration/targets/hcloud_floating_ip_info/tasks/test.yml

```diff
@@ -1,86 +1,77 @@
 
-- name: test gather hcloud floating ip infos in check mode
+- name: Gather hcloud_floating_ip_info in check mode
   hetzner.hcloud.hcloud_floating_ip_info:
   check_mode: true
-  register: hcloud_floating_ips
-
-- name: verify test gather hcloud floating ip infos in check mode
-  assert:
+  register: result
+- name: Verify hcloud_floating_ip_info in check mode
+  ansible.builtin.assert:
     that:
-      - hcloud_floating_ips.hcloud_floating_ip_info| list | count >= 1
+      - result.hcloud_floating_ip_info | list | count >= 1
 
-- name: test gather hcloud floating ip infos with correct label selector
+- name: Gather hcloud_floating_ip_info with correct id
   hetzner.hcloud.hcloud_floating_ip_info:
-    label_selector: "key=value"
-  register: hcloud_floating_ips
-- name: verify test gather hcloud floating ip with correct label selector
-  assert:
+    id: "{{ test_floating_ip.hcloud_floating_ip.id }}"
+  register: result
+- name: Verify hcloud_floating_ip_info with correct id
+  ansible.builtin.assert:
     that:
-      - hcloud_floating_ips.hcloud_floating_ip_info|selectattr('name','equalto','{{ test_floating_ip.hcloud_floating_ip.name }}') | list | count == 1
+      - result.hcloud_floating_ip_info | list | count == 1
 
-- name: test gather hcloud floating ip infos with wrong label selector
+- name: Gather hcloud_floating_ip_info with wrong id
   hetzner.hcloud.hcloud_floating_ip_info:
-    label_selector: "key!=value"
-  register: hcloud_floating_ips
-- name: verify test gather hcloud floating ip with wrong label selector
-  assert:
+    id: "{{ test_floating_ip.hcloud_floating_ip.id }}4321"
+  ignore_errors: true
+  register: result
+- name: Verify hcloud_floating_ip_info with wrong id
+  ansible.builtin.assert:
     that:
-      - hcloud_floating_ips.hcloud_floating_ip_info | list | count == 0
+      - result is failed
 
-- name: test gather hcloud floating ip infos with correct id
-  hetzner.hcloud.hcloud_floating_ip_info:
-    id: "{{test_floating_ip.hcloud_floating_ip.id}}"
-  register: hcloud_floating_ips
-- name: verify test gather hcloud floating ip with correct id
-  assert:
-    that:
-      - hcloud_floating_ips.hcloud_floating_ip_info|selectattr('name','equalto','{{ test_floating_ip.hcloud_floating_ip.name }}') | list | count == 1
+# - name: Gather hcloud_floating_ip_info with correct name
+#   hetzner.hcloud.hcloud_floating_ip_info:
+#     name: "{{ hcloud_floating_ip_name }}"
+#   register: result
+# - name: Verify hcloud_floating_ip_info with correct name
+#   ansible.builtin.assert:
+#     that:
+#       - result.hcloud_floating_ip_info | list | count == 1
+
+# - name: Gather hcloud_floating_ip_info with wrong name
+#   hetzner.hcloud.hcloud_floating_ip_info:
+#     name: "{{ hcloud_floating_ip_name }}-invalid"
+#   register: result
+# - name: Verify hcloud_floating_ip_info with wrong name
+#   ansible.builtin.assert:
+#     that:
+#       - result.hcloud_floating_ip_info | list | count == 0
```

Comment:
Why are these commented out?

Reply:
Because we don't implement it yet https://github.com/ansible-collections/hetzner.hcloud/issues/302

I have a branch with the implementation ready, just waiting on this PR to be merged to open a new PR.